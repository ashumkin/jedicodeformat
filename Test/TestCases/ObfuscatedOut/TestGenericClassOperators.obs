program testgenericclassoperators;{$APPTYPE CONSOLE}uses sysutils;type toperateur
<t> =class champ1:t;procedure test(const value:toperateur<t>);class operator negative
(const value:toperateur<t>):toperateur<t>;end;tmonrecord1=record procedure test(const
value:tmonrecord1);class operator negative(const value:tmonrecord1):tmonrecord1;end
;tmonrecord<t> =record champ1:t;procedure test(const value:tmonrecord<t>);class operator
negative(const value:tmonrecord<t>):tmonrecord<t>;end;end. 