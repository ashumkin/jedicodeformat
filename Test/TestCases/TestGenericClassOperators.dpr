program TestGenericClassOperators;

{$APPTYPE CONSOLE}


{ AFS December 2007 

  This code compiles, but is not semantically meaningfull.
  It is test cases for the code-formating utility

  Test new generics syntax - code from TridenT
}


uses
  SysUtils;

type
  //Ok
 TOperateur<T> = class
    Champ1: T;
    Procedure Test(const Value: TOperateur<T>);
    class operator Negative(const Value: TOperateur<T>): TOperateur<T>;
 end;
  //Ok
 TMonRecord1 = record
  Procedure Test(const Value: TMonRecord1);
  class operator Negative(const Value: TMonRecord1): TMonRecord1;
 end;

   //NOk
 TMonRecord<T> = record
    Champ1: T;
    Procedure Test(const Value: TMonRecord<T>);  //E2086 Le type 'TMonRecord<T>' n'est pas encore complètement défini
     //Opérateur Unaire -
    class operator Negative(const Value: TMonRecord<T>): TMonRecord<T>;
 end;

end.
