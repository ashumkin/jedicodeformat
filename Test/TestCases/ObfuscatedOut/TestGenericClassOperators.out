program testgenericclassoperators;{$APPTYPE CONSOLE}

uses
  SysUtils;

type
  toperateur<t> = class
    champ1: t;
    procedure test(const Value: toperateur<t>);
    class operator negative(const Value: toperateur<t>): toperateur<t>;
  end;
  tmonrecord1 = record
    procedure test(const Value: tmonrecord1);
      class operator negative(const Value: tmonrecord1): tmonrecord1;
  end;

  tmonrecord<t> = record
    champ1: t;
    procedure test(const Value: tmonrecord<t>);
      class operator
        negative(const Value: tmonrecord<t>): tmonrecord<t>;
  end;
end.
 